src/Editor/GUI/Editors/AnimationControllerEditor.cpp

```diff
@@ -52,6 +53,22 @@ void AnimationControllerEditor::ShowContextMenu() {
         memcpy(newTransition->name, name.c_str(), size);
         animationController->animationNodes.push_back(newTransition);
     }
+
+    ImGui::Separator();
+
+    if (ImGui::MenuItem("Add bool")) {
+        Animation::AnimationController::BoolItem* newBool = new Animation::AnimationController::BoolItem;
+        std::string name = "NewBool" + std::to_string(animationController->boolMap.size() + 1);
+        memcpy(newBool->name, name.c_str(), name.size() + 1);
```

Comment:
There's no guarantee that the destination is of sufficient capacity.

Reply:
Yes there is.

Reply:
Well unless the number of bools added is a Googol.